packages/@aws-cdk/aws-lambda-destinations/lib/chain.ts

```diff
@@ -0,0 +1,77 @@
+import * as events from '@aws-cdk/aws-events';
+import * as targets from '@aws-cdk/aws-events-targets';
+import * as lambda from '@aws-cdk/aws-lambda';
+import { Construct } from '@aws-cdk/core';
+import { EventBridgeDestination } from './event-bridge';
+
+/**
+ * Properties for a LambdaChain
+ */
+export interface LambdaChainProps {
+  /**
+   * The functions to chain.
+   *
+   * The functions cannot already have an asynchronous invocation
+   * configuration.
+   */
+  readonly functions: lambda.IFunction[];
+
+  /**
+   * The event bus to use.
+   *
+   * @default - default event bus
+   */
+  readonly eventBus?: events.IEventBus;
+
+  /**
+   * The destination for failed invocations.
+   *
+   * This destination will not be applied to the last function of the chain.
+   *
+   * @default - no destination
+   */
+  readonly onFailure?: lambda.IDestination;
+}
+
+/**
+ * Creates a chain of asynchronous Lambda invocations.
+ *
+ * The response payload of a **successful** invocation of a function in the
+ * chain is passed as request payload to the next function automatically using
+ * EventBridge.
+ */
+export class LambdaChain extends Construct {
+  constructor(scope: Construct, id: string, props: LambdaChainProps) {
+    super(scope, id);
+
+    let previousFn = props.functions[0];
+
+    for (const [idx, fn] of props.functions.entries()) {
+      // Configure destinations for all functions except the last one
+      if (idx < props.functions.length - 1) {
+        fn.configureAsyncInvoke({
+          onFailure: props.onFailure,
+          onSuccess: new EventBridgeDestination(props.eventBus),
```

Comment:
This is the trick here, it allows to extract via a rule the response payload (`$.detail.responsePayload`) so that the next function receives **exactly** the output of the previous function. This creates the abstraction.

With LambdaDestination you would have to extract the response payload in the runtime code and your Lambda function becomes "chain" specific.

It even allows to call the same lambda function as destination which is not possible with a Lambda destination. Tricky pattern though...

Reply:
Can't we somehow add this capability to `LambdaDestination` so that it will be possible to simply chain lambdas without a dedicated construct for it?

Reply:
or as another `Destination` class.